specifications/move_adapter/README.md

```diff
@@ -260,10 +276,15 @@ Move VM with gas metering disabled. Each kind of transaction payload has a
 corresponding prologue function that is used for validation. These prologue
 functions are defined in the `DiemAccount` module of the Diem Framework:
 
-* `Script`: The prologue function is `script_prologue`. In addition to the
-common checks listed below, it also calls the `is_script_allowed` function in
-the `DiemTransactionPublishingOption` module with the hash of the script
-bytecode to check if it is on the list of allowed scripts. If not, validation
+* `ScriptFunction` and `Script`: The prologue function is `script_prologue`.
+In addition to the common checks listed below, it also calls the `is_script_allowed`
+function in the `DiemTransactionPublishingOption` module to determine if the script
+should be allowed. A script sent by an account with `has_diem_root_role` is always
+allowed. Otherwise, a `Script` payload is allowed if the hash of the
+script bytecode is on the list of allowed scripts published at
+`0x1::DiemTransactionPublishingOption::DiemTransactionPublishingOption.script_allowlist`.
+`ScriptFunction` payloads, for which the adapter uses an empty vector in place
+of the script hash, are always allowed. If the script is not allowed, validation
```

Comment:
It's hard to tell if this is referring to only scripts, or also script functions here. Might be useful to clarify what the status code is if the script function doesn't exist.

Reply:
I don't think there is any check for whether a script function exists during validation. That check is done later by the VM during the execution phase. I added two sentences later in the VM section to describe the specific error codes for the cases where the function does not exist or does not have `script` visibility.